starter_code/elevator.js

```diff
@@ -3,17 +3,63 @@ class Elevator {
     this.floor      = 0;
     this.MAXFLOOR   = 10;
     this.requests   = [];
+    this.waitingList = [];
+    this.passengers = [];
+    this.direction = "up";
+    this.setInterval;
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start() {
+  this.setInterval = setInterval(() => {
+    this.update();
+  }, 1000);
+  }
+
+  stop() {
+    clearInterval(this.setInterval);
+  }
+  update() {
+this.log();
+  }
+  _passengersEnter(person) {
+if(!this.waitingList.length == 0){
+  this.passengers.push(person);
+  this.waitingList.pop();
+  requests.destinationFloor.push(person.originFloor);
+  console.log(person.name + "has enter the elevator");
+}
+  }
+  _passengersLeave(person) {
+    if(!this.passengers == 0){
+      if (person.destinationFloor == this.floor){
+        this.passengers.pop(person)
+      }
+      console.log(person.name + "has left the elevator");
+    }
+
+   }
+
+  floorUp() {
+  if(this.floor < this.MAXFLOOR){
+  this.floor ++;
+  console.log(this.log());
```

Comment:
No tienes que utilizar console.log , solo tienes q llamar a this.log (esa funcion es la que se encarga de decirte en que posicion está el ascensor)